crates/holochain/src/core/ribosome/guest_callback/validate.rs

```diff
@@ -276,6 +278,44 @@ mod slow_tests {
         assert_eq!(result, ValidateResult::Valid,);
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_validate_implemented_invalid_return() {
```

Comment:
Cool, so for this PR
- Can we have two tests with the new zomes you've added but rather than calling `validate` directly, create some data in the coordinator and let Holochain call the `validate` callback and make sure we get the right errors back

For a future PR
- We should look at inline app validation and what action to take when we get this error. I suspect shutting down the app is the right action there but we should discuss